### ISSUE TITLE
ebiten: add a new API to return the logical screen size (= the returns of Layout)

### ISSUE THREAD

**Issue by mcarpenter622** [hajimehoshi/ebiten#2986]
### Operating System

- [X] Windows
- [X] macOS
- [X] Linux
- [X] FreeBSD
- [X] OpenBSD
- [x] Android
- [x] iOS
- [x] Nintendo Switch
- [x] PlayStation 5
- [x] Xbox
- [x] Web Browsers

### What feature would you like to be added?

I would like a new function or enhancement to GetScreenSize that gives the proper size of the viewable area regardless of if it is maximized or not.

It would be ideal of this new function worked in all environments.

It feels clunky to have to write code like this to get the actual viewable area:

```go
ww, wh := ebiten.WindowSize()
    if ebiten.IsFullscreen() {
        ww, wh = ebiten.ScreenSizeInFullscreen()
    }
```

(I know this is a little outdated and should use Monitor().Size() )

### Why is this needed?

To simplify the development experience.

**Comment by hajimehoshi** [comment]
> GetWindowSize alternative that takes into account if the window is maximized

Maximized window and fullsceening are different things. Do you mean you want to get a monitor size at WindowSize when fullscreening?

**Comment by mcarpenter622** [comment]
I meant FullScreen mode. I would like a function to get the viewable size that ebitengine is drawing on, regardless of window state. Essentially, I'm trying to write code to center something on the screen and I have to use that snippet above to make it work properly whether the game is in FullScreen mode or not. Ideally this new function would work for mobile devices too.

**Comment by hajimehoshi** [comment]
You can keep the arguments of `Layout` function and use it in `Update`.

> Ideally this new function would work for mobile devices too.

Mobiles don't have a window.

**Comment by hajimehoshi** [comment]
Good point. I hesitated to add a new global function that are available only when the game is running. Let me think.

**Comment by hajimehoshi** [comment]
> I would like a new function or enhancement to GetScreenSize that gives the proper size of the viewable area regardless of if it is maximized or not.

I understand your proposal `GetScreenSize` would be nice in terms of consistency, but I was wondering what an actual use case would be.

**Comment by mcarpenter622** [comment]
The use case is that I am trying to center a dialog within the window, but because ebiten.WindowSize() doesnt take FullScreen into account, I have to write the code like this to work regardless of if the game is in windowed mode or full screen mode, which was a bit of a "gotcha"
```
	ww, wh := ebiten.WindowSize()
	if ebiten.IsFullscreen() {
		ww, wh = ebiten.ScreenSizeInFullscreen()
	}
```

**Comment by hajimehoshi** [comment]
> The use case is that I am trying to center a dialog within the window,

Which Ebitengine or the OS renders your dialog? If Ebitengine does, what you need is the result of Layout rather than the attributes of Layout, right? If the OS does, the unit (Ebitengine's device independent pixels) might be different from the OS so this is a little tricky.

**Comment by hajimehoshi** [comment]
So your proposal

```go
func GetScreenSize() (int, int)
	ww, wh := ebiten.WindowSize()
	if ebiten.IsFullscreen() {
		ww, wh = ebiten.ScreenSizeInFullscreen()
	}
	return ww, wh
}
```

works only when your Layout is an identity function returning the arguments without modification.

In general, if you want to get the position to render something center in the game screen, you should want the logical screen size = the size of the screen image, not the window size.

If so, I understand the function returning the logical screen size, not the window size, would be useful. I just would like to confirm we are on the same page.

**Comment by mcarpenter622** [comment]
Ebitengine is rendering it. Yes we are on the same page.

**Comment by hajimehoshi** [comment]
Thanks!

**Comment by hajimehoshi** [comment]
I believe this is not a critical feature, so let me postpone the milestone.

**Comment by scratch85** [comment]
I would appreciate if such a functionality would work on all devices and browser. The proposed function will not work on mobile or in wasm.

**Comment by hajimehoshi** [comment]
Doesn't caching layout results by yourself work?

**Comment by scratch85** [comment]
> Doesn't recording layout results by yourself work?

If this is the recommended way, that is fine for me. Like:

```golang
func (g *Game) Layout(width, height int) (int, int) {
	g.width = width
	g.height = height 
	return width, height
}
```

**Comment by hajimehoshi** [comment]
Yeah, this is the current recommended way.
